internal/metrics/metrics.go

```diff
@@ -0,0 +1,89 @@
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics holds all prometheus metrics
+type Metrics struct {
+	BackendLatency    *prometheus.HistogramVec
+	TokensTotal       *prometheus.CounterVec
+	RequestsTotal     *prometheus.CounterVec
+	FirstTokenLatency *prometheus.HistogramVec
+	InterTokenLatency *prometheus.HistogramVec
+	Registry          *prometheus.Registry
+}
+
+var (
+	instance *Metrics
+	once     sync.Once
+)
+
+// New creates a new Metrics instance
+func New() *Metrics {
```

Comment:
Make this private so everyone calls `GetOrCreate()`

Reply:
updated